net/rhp/session.go

```diff
@@ -100,10 +100,7 @@ func DialSession(conn net.Conn, pub types.PublicKey) (_ *Session, err error) {
 		}
 	}()
 	// exchange versions and read host's initial challenge
-	s, err := m.DialStream()
-	if err != nil {
-		return nil, err
-	}
+	s := m.DialStream()
 	defer s.Close()
```

Comment:
I'm definitely on the "ignoring `Close()` errors is fine" end of the spectrum. There are exceptions, obviously (e.g. `zip.Close` writes a trailer to the underlying stream, and you definitely want to know if that failed), but in the vast majority of cases, the error isn't useful or actionable.

For this case in particular, `Close` makes a "best-effort" attempt to inform the other party that we're closing the stream. If that attempt fails, it's either because the other party already closed the stream, or because the underlying connection has been broken. In the former case, we don't care, and in the latter case, the error will be handled when we try to `Read`, `Write`, or `Accept` a stream.